sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -96,35 +99,16 @@ static struct snd_soc_card hda_soc_card = {
 	.late_probe = skl_hda_card_late_probe,
 };
 
-#define IDISP_DAI_COUNT		3
-/* there are two routes per iDisp output */
-#define IDISP_ROUTE_COUNT	(IDISP_DAI_COUNT * 2)
-#define IDISP_CODEC_MASK	0x4
-
 static int skl_hda_fill_card_info(struct snd_soc_acpi_mach_params *mach_params)
 {
 	struct snd_soc_card *card = &hda_soc_card;
 	struct snd_soc_dai_link *dai_link;
-	u32 codec_count, codec_mask;
-	int i, num_links, num_route;
-
-	codec_mask = mach_params->codec_mask;
-	codec_count = hweight_long(codec_mask);
-
-	if (codec_count == 1 && codec_mask & IDISP_CODEC_MASK) {
-		num_links = IDISP_DAI_COUNT;
-		num_route = IDISP_ROUTE_COUNT;
-	} else if (codec_count == 2 && codec_mask & IDISP_CODEC_MASK) {
-		num_links = ARRAY_SIZE(skl_hda_be_dai_links);
-		num_route = ARRAY_SIZE(skl_hda_map),
-		card->dapm_widgets = skl_hda_widgets;
-		card->num_dapm_widgets = ARRAY_SIZE(skl_hda_widgets);
-	} else {
-		return -EINVAL;
-	}
+	int i;
 
-	card->num_links = num_links;
-	card->num_dapm_routes = num_route;
+	card->num_links = ARRAY_SIZE(skl_hda_be_dai_links);
+	card->num_dapm_routes = ARRAY_SIZE(skl_hda_map),
+	card->num_dapm_widgets = ARRAY_SIZE(skl_hda_widgets);
+	card->dapm_widgets = skl_hda_widgets;
```

Comment:
you are changing the functionality here. If there is no hdaudio codec, then we only want the idisp dailinks. And you need to keep testing for the codec count, we can only do 1 or 2 codecs.

Reply:
please test this on Up2 with HDMI+ DMIC.

Reply:
yes, I misunderstood the algorithm in ASOC. For Up2, A error of no codec dai link would be reported.